### PR TITLE
Add in squared loss for regression in linear model

### PR DESCRIPTION
**Pull request by hectorgon**
Squared loss regression
@deerzq @yolken @spencerde

### CODE REVIEW

File: training/src/main/scala/com/airbnb/aerosolve/training/LinearRankerTrainer.scala [hunk 1]
```diff
@@ -75,6 +75,7 @@ object LinearRankerTrainer {
     loss match {
       case "ranking" => rankingTrain(sc, input, config, key, numBags, weights, iteration)
       case "regression" => regressionTrain(sc, input, config, key, numBags, weights, iteration)
+      case "regression2" => regression2Train(sc, input, config, key, numBags, weights, iteration)
```
Comment (deerzq) [inline training/src/main/scala/com/airbnb/aerosolve/training/LinearRankerTrainer.scala]: `regression2` sounds not so informative

File: training/src/main/scala/com/airbnb/aerosolve/training/LinearRankerTrainer.scala [hunk 2]
```diff
@@ -149,6 +150,66 @@ object LinearRankerTrainer {
...
+                                        sum = newGradSum)
+            weightMap.put(v, (newWeight, newGradSum))
+          })
+          weightMap.put(lossKey, (lossEntry._1 + loss, lossEntry._2 + 1.0))          
+        })
+      })
+      weightMap
+        .iterator
+    })
+  }
```
Comment (deerzq) [inline training/src/main/scala/com/airbnb/aerosolve/training/LinearRankerTrainer.scala]: can we ask the user to pass in regression loss function name in config, then modify regressionTrain to take different updates based on the loss function name?

File: training/src/main/scala/com/airbnb/aerosolve/training/LinearRankerTrainer.scala (hunk 1 above)
Comment (hectorgon) [inline training/src/main/scala/com/airbnb/aerosolve/training/LinearRankerTrainer.scala]: Fixed

File: training/src/main/scala/com/airbnb/aerosolve/training/LinearRankerTrainer.scala (hunk 2 above)
Comment (hectorgon) [inline training/src/main/scala/com/airbnb/aerosolve/training/LinearRankerTrainer.scala]: It's not that much code and then we'd have problems with epsilon which we don't require for L2 so I'd rather keep it as a separate function instead of having case statements inside the loop
